### DISCUSSION TITLE
Rate Limit Based on JWT Claims

### DISCUSSION THREAD

**Discussion by pleshakov** [nginx/kubernetes-ingress#4911] [Ideas]
If a user configures a JWT Policy https://docs.nginx.com/nginx-ingress-controller/configuration/policy-resource/#jwt-using-local-kubernetes-secret and also a rate limit policy https://docs.nginx.com/nginx-ingress-controller/configuration/policy-resource/#ratelimit , currently, it is not possible to use JWT claims (like sub) as a rate limiting key.

Supporting rate limiting based on JWT claims will enable more granural rate limiting -- it will be possible to use information about an authenticated user.

Here is an example of such rate limiting in the native NGINX Plus configurtion -- https://www.nginx.com/blog/authenticating-api-clients-jwt-nginx-plus/#Leveraging-JWT-Claims-for-Logging-and-Rate-Limiting

Related: https://github.com/nginxinc/kubernetes-ingress/discussions/4417

**Comment by pdabelf5** [comment]
https://docs.nginx.com/nginx-ingress-controller/releases/#500
